mongo/integration/client_test.go

```diff
@@ -472,3 +474,108 @@ type proxyMessage struct {
 	sent          wiremessage.WireMessage
 	received      wiremessage.WireMessage
 }
+
+func TestClientStress(t *testing.T) {
+	// TODO: Enable with GODRIVER-2038.
+	t.Skip("TODO: Enable with GODRIVER-2038")
+
+	mtOpts := mtest.NewOptions().
+		MinServerVersion("3.6").
+		Topologies(mtest.ReplicaSet, mtest.Sharded, mtest.Single).
+		CreateClient(false)
+	mt := mtest.New(t, mtOpts)
+
+	// Test that a Client can recover from a massive traffic spikes after the traffic spike is over.
+	mt.Run("Client recovers from traffic spike", func(mt *mtest.T) {
+		oid := primitive.NewObjectID()
+		doc := bson.D{{Key: "_id", Value: oid}, {Key: "key", Value: "value"}}
+		_, err := mt.Coll.InsertOne(context.Background(), doc)
+		assert.Nil(mt, err, "unexpected error inserting document: %v", err)
```

Comment:
I think error messages from methods/functions should be of the format `InsertOne error: %v` here and below.

Reply:
Will update.

Reply:
Updated.